lab8/src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -0,0 +1,67 @@
+package agh.ics.oop;
+
+import agh.ics.oop.gui.AppVisualizer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SimulationEngine implements IEngine, Runnable{
+    private MoveDirection[] moveDirections;
+    private final IWorldMap map;
+    private final Vector2d[] animalsVectors;
+    private final List<Animal> animalsOnMap = new ArrayList<>();
+    protected AppVisualizer appVisualizer;
```

Comment:
tej zależności zdecydowanie nie chcemy, właśnie po to nam observer żeby uniknąć sytuacji gdy engine musi znać GUI. Wystarczyłoby żeby konstruktor przyjmował go jako `IPositionChangeObserver` i nie zapisywał tego w atrybucie nawet, tylko rejestrował od razu w zwierzątkach.